### PR TITLE
test: fs: improve code coverage for file system core

### PR DESCRIPTION
**Pull request by mengxianglinx**
Refine test cases to improve code coverage
Refine comments for each test case

Signed-off-by: Meng xianglin <xianglinx.meng@intel.com>

### CODE REVIEW

File: tests/subsys/fs/fs_api/src/test_multi_fs.c [hunk 1]
```diff
@@ -69,6 +74,11 @@ static int test_fs_readmount(void)
 
 static int test_fs_deinit(void)
 {
+	/* NULL parameter */
+	if (fs_unregister(TEST_FS_1, NULL) == 0) {
+		return TC_FAIL;
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_multi_fs.c:79]: Use here `-EINVAL` as in all other cases; or convert all other cases to `TC_FAIL`.

File: tests/subsys/fs/fs_api/src/test_multi_fs.c [hunk 2]
```diff
@@ -105,12 +115,39 @@ static int test_fs_external(void)
...
+ * - dynamic analysis and testing
+ *
+ * Prerequisite Conditions:
+ * - N/A
+ *
+ * Input Specifications:
+ * - N/A
+ *
+ * Test Procedure:
+ * -# regiter two filesystems and mount them
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_multi_fs.c:135]: regiter->register

File: tests/subsys/fs/fs_api/src/test_multi_fs.c [hunk 3]
```diff
@@ -30,6 +30,11 @@ static int test_fs_init(void)
 		return -EINVAL;
 	}
 
+	/* re-register a file system */
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_multi_fs.c:33]: "Attempt to re-register (...)"

File: tests/subsys/fs/fs_api/src/test_multi_fs.c [hunk 4]
```diff
@@ -105,12 +115,39 @@ static int test_fs_external(void)
...
  * @brief Multi file systems register and unregister
  *
- * @details register and unregister two file systems to test
- *          the system support multiple file system simultanously
+ * @details
+ * Test Objective:
+ * - register and unregister two file systems to test the system support
+ *   multiple file system simultanously
+ *
+ * Testing techniques:
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_multi_fs.c:123]: You probably want to capitalize the "techniques", as you do with all words of titles in all other sections.

File: tests/subsys/fs/fs_api/src/test_multi_fs.c [hunk 5]
```diff
@@ -105,12 +115,39 @@ static int test_fs_external(void)
...
+ *
+ * Prerequisite Conditions:
+ * - N/A
+ *
+ * Input Specifications:
+ * - N/A
+ *
+ * Test Procedure:
+ * -# regiter two filesystems and mount them
+ * -# check how many filesystems have been rigistered via fs_readmount()
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_multi_fs.c:136]: `fs_readmount` does not iterate through the list of registered file systems but mounted file systems.

File: tests/subsys/fs/fs_api/src/test_multi_fs.c [hunk 6]
```diff
@@ -105,12 +115,39 @@ static int test_fs_external(void)
...
+ * -# unmount and unregister all filesystems
+ *
+ * Expected Test Result:
+ * - Filesystems are mounted and unmouted properly
+ *
+ * Pass/Fail Criteria:
+ * - Successful if filesystems are mounted and unmounted as expected,
+ *   failure otherwise
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_multi_fs.c:144]: Isn't the "Expected Test Result" and "Pass/Fail Criteria" redundant?

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 7]
```diff
@@ -94,17 +152,53 @@ void test_mount(void)
...
+ * Prerequisite Conditions:
+ * - N/A
+ *
+ * Input Specifications:
+ * - N/A
+ *
+ * Test Procedure:
+ * -# Unmount a file system with NULL parameter
+ * -# Unmount file system that has never been mounted
+ * -# Unmount a file system
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:189]: A file system that has previously been mounted?

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 8]
```diff
@@ -94,17 +152,53 @@ void test_mount(void)
...
+ * -# Reunmount a file system
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:190]: Reword it to something like "Once attempt to unmount the file system from previous step/scenario"

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 9]
```diff
@@ -94,17 +152,53 @@ void test_mount(void)
...
+ * -# Unmount a unmountable file system
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:191]: unmount file system that can not be mounted?

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 10]
```diff
@@ -32,36 +32,89 @@ static struct fs_mount_t test_fs_mnt_unsupported_fs = {
...
+ * -# Mount a file system which has no mount interface
+ * -# Mount a file system which has no unmount interface
+ *
+ * Expected Test Result:
+ * - Step 6, 8 pass and other steps return error
+ *
+ * Pass/Fail Criteria:
+ * - Successful if every case pass as expected, failure otherwise
+ *
+ * Assumptions and Constraints:
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:115]: This sub-section never gets filled. I do not think it is needed it just bloats documentation.

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 11]
```diff
@@ -32,36 +32,89 @@ static struct fs_mount_t test_fs_mnt_unsupported_fs = {
...
+ * Testing techniques:
+ * - Interface testing
+ * - function and block box testing
+ * - dynamic analysis and testing
+ *
+ * Prerequisite Conditions:
+ * - N/A
+ *
+ * Input Specifications:
+ * - N/A
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:97]: What is a reason for including subsections that have no content?

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 12]
```diff
@@ -73,14 +126,19 @@ void test_mount(void)
 	ret = fs_mount(NULL);
 	zassert_not_equal(ret, 0, "Mount a NULL fs");
 
-	TC_PRINT("Mount to a unsupported directory\n");
+	TC_PRINT("Mount a unsupported file system\n");
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:129]: "an unsupported"

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 13]
```diff
@@ -73,14 +126,19 @@ void test_mount(void)
...
 	zassert_not_equal(ret, 0, "Mount a unsupported fs");
 
 	fs_register(TEST_FS_2, &temp_fs);
+
 	TC_PRINT("Mount to an invalid directory\n");
-	ret = fs_mount(&test_fs_mnt_invalid_root);
+	ret = fs_mount(&test_fs_mnt_invalid_root_1);
 	zassert_not_equal(ret, 0, "Mount to an invalid dir");
+	ret = fs_mount(&test_fs_mnt_invalid_root_2);
+	zassert_not_equal(ret, 0, "Mount dir has too short name");
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:139]: "dir name too short";

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 14]
```diff
@@ -94,17 +152,53 @@ void test_mount(void)
...
 
 	fs_unregister(TEST_FS_2, &temp_fs);
-
 	fs_register(TEST_FS_2, &null_fs);
+
 	TC_PRINT("Mount a file system has no interface implemented\n");
 	ret = fs_mount(&test_fs_mnt_no_op);
 	zassert_not_equal(ret, 0, "Mount to a fs without op interface");
-	fs_unregister(TEST_FS_2, &null_fs);
+	ret = fs_unmount(&test_fs_mnt_no_op);
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:160]: I think you wanted to do something with the `ret`.

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 15]
```diff
@@ -94,17 +152,53 @@ void test_mount(void)
...
+
+	/* mount an unmountable file system */
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:162]: "in-unmountable"?? better reword somehow.

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 16]
```diff
@@ -94,17 +152,53 @@ void test_mount(void)
...
+	null_fs.mount = temp_fs.mount;
+	ret = fs_mount(&test_fs_mnt_no_op);
+	zassert_equal(ret, 0, "Mount to a fs without unmount interface");
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:165]: "interface"->"functionality"; interface is the entire FS API provided by fs driver.

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 17]
```diff
@@ -119,20 +213,56 @@ void test_unmount(void)
...
 
-	TC_PRINT("\nunmount file system multiple times:\n");
 	ret = fs_unmount(&test_fs_mnt_1);
 	zassert_true(ret >= 0, "Fail to unmount fs");
 
+	TC_PRINT("\nunmount file system multiple times:\n");
 	test_fs_mnt_1.fs = &temp_fs;
 	ret = fs_unmount(&test_fs_mnt_1);
-	zassert_equal(ret, -EINVAL, "Unmount a unmounted fs");
+	zassert_not_equal(ret, 0, "Unmount a unmounted fs");
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:224]: "an unmounted"; usually 'an' before 'u'.

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 18]
```diff
@@ -119,20 +213,56 @@ void test_unmount(void)
...
+
+	TC_PRINT("unmount an unmountable file system\n");
+	ret = fs_unmount(&test_fs_mnt_no_op);
+	zassert_not_equal(ret, 0, "Unmount a unmountable fs");
+	/* TEST_FS_2 is registered in test_mount(), unregister it here*/
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:229]: missing space after here

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 19]
```diff
@@ -119,20 +213,56 @@ void test_unmount(void)
...
  * @brief Test statvfs interface of filesystem
  *
+ * @details
+ * Test Objective
+ * - Test fs_statvfs() interface in file system
+ *
+ * Testing techniques:
+ * - Interface testing
+ * - function and block box testing
+ * - dynamic analysis and testing
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:243]: I do not understand the claim.

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 20]
```diff
@@ -142,16 +272,24 @@ void test_file_statvfs(void)
 
 	ret = fs_statvfs(NULL, &stat);
 	zassert_not_equal(ret, 0, "Get valume without path");
+	ret = fs_statvfs(TEST_FS_MNTP, NULL);
+	zassert_not_equal(ret, 0, "Get valume without stat structure");
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:276]: I think that the "value" has been misspelled in line 274 and the mistake has been carried here.
Why not just "pass NULL for stat structure pointer"?

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 21]
```diff
@@ -142,16 +272,24 @@ void test_file_statvfs(void)
...
+
+	ret = fs_statvfs("/", &stat);
+	zassert_not_equal(ret, 0, "Get valume by too short path");
+
+	ret = fs_statvfs("SDCARD:", &stat);
+	zassert_not_equal(ret, 0, "Get valume by path without /");
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:282]: In neither of these you are attempting to "Get valume [sic]", you are attempting to get volume statistics, by the function called.

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 22]
```diff
@@ -142,16 +272,24 @@ void test_file_statvfs(void)
...
 
 	ret = fs_statvfs("/SDCARD:", &stat);
 	zassert_not_equal(ret, 0, "Get valume by no-exist path");
 
-	ret = fs_statvfs(TEST_FS_MNTP, NULL);
-	zassert_not_equal(ret, 0, "Get valume without stat structure");
+	/* It's ok if there is no stat interface */
+	ret = fs_statvfs(NOOP_MNTP, &stat);
+	zassert_equal(ret, 0, "Get valume without stat interface");
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:289]: There is an interface, it lacks the function implementation.

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 23]
```diff
@@ -32,36 +32,89 @@ static struct fs_mount_t test_fs_mnt_unsupported_fs = {
...
 static struct fs_file_t filep;
+static struct fs_file_t err_filep;
 static const char test_str[] = "hello world!";
 
 /**
  * @brief Test mount interface of filesystem
  *
+ * @details
+ * Test Objective
+ * - Test fs_mount() interface in file system
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:85]: Are you testing interface of the `fs_mount` function?

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 24]
```diff
@@ -32,36 +32,89 @@ static struct fs_mount_t test_fs_mnt_unsupported_fs = {
...
+ * - Following test cases depend on file systems mounted in this test case
+ *
+ * Testing techniques:
+ * - Interface testing
+ * - function and block box testing
+ * - dynamic analysis and testing
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:91]: What is "dynamic analysis"? Why is the "testing" testing technique?

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 25]
```diff
@@ -32,36 +32,89 @@ static struct fs_mount_t test_fs_mnt_unsupported_fs = {
...
+ *
+ * Prerequisite Conditions:
+ * - N/A
+ *
+ * Input Specifications:
+ * - N/A
+ *
+ * Test Procedure:
+ * -# Mount a file system with NULL parameter
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:100]: What parameter?

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 26]
```diff
@@ -32,36 +32,89 @@ static struct fs_mount_t test_fs_mnt_unsupported_fs = {
...
+ * -# Mount a file system unsupported
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:101]: Did you mean "Attempt mounting an unsupported file system"?

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 27]
```diff
@@ -32,36 +32,89 @@ static struct fs_mount_t test_fs_mnt_unsupported_fs = {
...
+ * -# Register a file system and mount it to invalide directories
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:102]: "invalide" -> "invalid"; "attempt mounting it at invalid directory paths" maybe...

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 28]
```diff
@@ -32,36 +32,89 @@ static struct fs_mount_t test_fs_mnt_unsupported_fs = {
...
+ * -# Pass invalid parameter to specific file system interface
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:103]: What is interface? Function? What parameter will be invalid, any?

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 29]
```diff
@@ -32,36 +32,89 @@ static struct fs_mount_t test_fs_mnt_unsupported_fs = {
...
+ * -# Mount a normal file system to a normal directory
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:104]: What is "normal" file system?
Do you mean "existing directory"?

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 30]
```diff
@@ -32,36 +32,89 @@ static struct fs_mount_t test_fs_mnt_unsupported_fs = {
...
+ * -# Remount a mounted file system
+ * -# Mount a file system which has no mount interface
+ * -# Mount a file system which has no unmount interface
+ *
+ * Expected Test Result:
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:109]: I am quite convinced that the usage of automatic numbering in the "Test Procedure" is bad idea if you are going to literally place numbers here; each time the procedure gets updated someone will have to recount numbers and update them.

Wouldn't it be better to list expected test result with a step in test procedure? For example:
```
- # Remount a mounted file system [expect to PASS]
```
or
```
- # [expect PASS] Remount a mounted file system
```
Or some better way?

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 31]
```diff
@@ -73,14 +126,19 @@ void test_mount(void)
...
 	fs_register(TEST_FS_2, &temp_fs);
+
 	TC_PRINT("Mount to an invalid directory\n");
-	ret = fs_mount(&test_fs_mnt_invalid_root);
+	ret = fs_mount(&test_fs_mnt_invalid_root_1);
 	zassert_not_equal(ret, 0, "Mount to an invalid dir");
+	ret = fs_mount(&test_fs_mnt_invalid_root_2);
+	zassert_not_equal(ret, 0, "Mount dir has too short name");
+	ret = fs_mount(&test_fs_mnt_invalid_root_3);
+	zassert_not_equal(ret, 0, "Mount to NULL mount point");
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:141]: "/NULL" ? or the pointer is NULL?

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 32]
```diff
@@ -94,17 +152,53 @@ void test_mount(void)
...
+	ret = fs_mount(&test_fs_mnt_no_op);
+	zassert_equal(ret, 0, "Mount to a fs without unmount interface");
 }
 
 /**
  * @brief Test unmount interface of filesystem
  *
+ * @details
+ * Test Objective
+ * - Test fs_unmount() interface in file system
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:173]: `fs_umount` is a function of an fs interface.

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 33]
```diff
@@ -119,20 +213,56 @@ void test_unmount(void)
...
+ * - N/A
+ *
+ * Input Specifications:
+ * - N/A
+ *
+ * Test Procedure:
+ * -# Stat valume with NULL parameter
+ * -# Stat valume with invalid path
+ * -# Stat a valume when the file system has no fs_statvfs interface
+ * -# Get valume infomation of a normal file system
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:255]: What are these "valumes"?

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 34]
```diff
@@ -202,27 +409,66 @@ void test_opendir(void)
...
+ * Prerequisite Conditions:
+ * - N/A
+ *
+ * Input Specifications:
+ * - N/A
+ *
+ * Test Procedure:
+ * -# Open a directiory then close it
+ * -# Close a closed or never opened directory
+ * -# Close a directory whoes file system has no close interface
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:461]: "whoes"->"whose"

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 35]
```diff
@@ -303,7 +559,37 @@ static int _test_lsdir(const char *path)
...
+ * Prerequisite Conditions:
+ * - N/A
+ *
+ * Input Specifications:
+ * - N/A
+ *
+ * Test Procedure:
+ * -# Read a NULL directory
+ * -# Read root directory
+ * -# Read non-exist directory
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:582]: "non-exist"->"not existing"

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c (hunk 24 above)
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:91]: Why this section is repeated everywhere? It is all the same it just bloats the documentation. Shouldn't it be mentioned somewhere in general section?

File: tests/subsys/fs/fs_api/src/test_multi_fs.c [hunk 36]
```diff
@@ -105,12 +115,39 @@ static int test_fs_external(void)
 /**
  * @brief Multi file systems register and unregister
  *
- * @details register and unregister two file systems to test
- *          the system support multiple file system simultanously
+ * @details
+ * Test Objective:
+ * - register and unregister two file systems to test the system support
+ *   multiple file system simultanously
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_multi_fs.c:121]: "simultanously"->"simultaneously"

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 37]
```diff
@@ -706,12 +1377,30 @@ void test_file_unlink(void)
 	ret = fs_unlink(NULL);
 	zassert_not_equal(ret, 0, "Delete a NULL file");
 
+	ret = fs_unlink("a");
+	zassert_not_equal(ret, 0, "Delete a file with too short name");
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:1381]: The name is not too short, it is missing mount point.

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 38]
```diff
@@ -706,12 +1377,30 @@ void test_file_unlink(void)
...
+
+	ret = fs_unlink("SDCARD:/test_file.txt");
+	zassert_not_equal(ret, 0, "Delete a file with wrong path");
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:1384]: How "wrong" is the path? Why  nor write "Attempt deleting file with missing root / in path".

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 39]
```diff
@@ -706,12 +1377,30 @@ void test_file_unlink(void)
...
 	ret = fs_unlink(TEST_FS_MNTP);
 	zassert_not_equal(ret, 0, "Delete a root dir");
 
+	ret = fs_unlink(NOOP_MNTP);
+	zassert_not_equal(ret, 0, "Filesystem has no unlink interface");
+
+	/* in filesystem core, function fs_get_mnt_point() check mountp_len
+	 * of all mounted mount point, this case is the last case before unmount,
+	 * change mountp_len of test_fs_mnt_no_op to 0 to test
+	 * function fs_get_mnt_point()
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:1398]: I don't get that at all.

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 40]
```diff
@@ -706,12 +1377,30 @@ void test_file_unlink(void)
...
+	zassert_not_equal(ret, 0, "Delete a file with wrong path");
+
 	ret = fs_unlink("/SDCARD:/test_file.txt");
 	zassert_not_equal(ret, 0, "Delete a non-exist file");
 
 	ret = fs_unlink(TEST_FS_MNTP);
 	zassert_not_equal(ret, 0, "Delete a root dir");
 
+	ret = fs_unlink(NOOP_MNTP);
+	zassert_not_equal(ret, 0, "Filesystem has no unlink interface");
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:1393]: `unlink` is a function, when it is `fs_unlink` or a functionality in general.

**Review by de-nordic — CHANGES_REQUESTED** [review]
The additional test cases are fine, but generally: no.

The test documentation provided is bloating documentation while not providing enough of a substantial value.
I could agree with the "Test Procedures", if they would consist of a higher level requirements or a rationale/description of what a tests is intended to achieve, with the actual code referencing such procedure description, by test case id or some other way.
Currently they are even more ambiguous than the code that implements them.

And there is a lot to fix.

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c (hunk 11 above)
Comment (mengxianglinx) [reply tests/subsys/fs/fs_api/src/test_fs_dir_file.c:97]: Organizing comments in such a format is for generating documents to adhere to the requirement of IEC-61508. For most test cases for fs_api, there is no "Prerequisite Conditions" or "Input Specifications", but in some test cases for other modules, there might be.

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c (hunk 19 above)
Comment (mengxianglinx) [reply tests/subsys/fs/fs_api/src/test_fs_dir_file.c:233]: These testing techniques are described in IEC-61508, in test report for every test case, the "testing techniques" need to be listed

Comment (de-nordic) [reply tests/subsys/fs/fs_api/src/test_fs_dir_file.c:233]: What I meant is that, as far as I understand, the "dynamic analysis" is form of testing itself so addition of "testing" does change it's nature; additionally the subsection is called "Testing Techniques" so listing "testing" as a method seems redundant.

Comment (de-nordic) [reply tests/subsys/fs/fs_api/src/test_fs_dir_file.c:233]: > 
> These testing techniques are described in IEC-61508, in test report for every test case, the "testing techniques" need to be listed

@mengxianglinx OK, so this is part of bigger effort. I am not familiar with the IEC-61508, and my brief scroll through the documents did not make me an expert, but shouldn't the entire form/structure of comments be discussed and defined prior to starting with the effort?
I guess that this is going to be implemented in all other tests.
Have other forms been considered and the effect of bloating the code taken into account?
Has there been a template provided, so that it could be used instead of copy-paste from other test cases?

Comment (mengxianglinx) [reply tests/subsys/fs/fs_api/src/test_fs_dir_file.c:233]: Thank you @de-nordic ! Our team is trying to simplify document form/structure to avoid bloating the code. Today we have an online meeting about Function Safety [test work group](https://github.com/zephyrproject-rtos/zephyr/wiki/Zephyr-Committee-and-Working-Group-Meetings#testing-working-group) to discuss about these comment formats. We are appreciated if you can attend this meeting.

**Comment by LeiW000** [comment]
@de-nordic , do you still think we should not add doxygen comments like what we did in this PR? If yes, we can remove it from this PR. Let's this PR focused on code of the test cases.

We plan to file a RFC for format of doxygen comment of test case to reach an agreement in the community. After that, we can add doxygen comments for the test cases with another PR.

**Comment by de-nordic** [comment]
> 
> 
> @de-nordic , do you still think we should not add doxygen comments like what we did in this PR? If yes, we can remove it from this PR. Let's this PR focused on code of the test cases.
> 
> We plan to file a RFC for format of doxygen comment of test case to reach an agreement in the community. After that, we can add doxygen comments for the test cases with another PR.

@LeiW000, @mengxianglinx yes, please do that.  Focus on improvements to the test cases so that we could quickly merge them in.

**Review by LeiW000 — APPROVED** [review]

**Comment by LeiW000** [comment]
@de-nordic , Please take look at it again. Xianglin has updated the PR to remove unecessary doxygen comments.

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 41]
```diff
@@ -332,9 +401,21 @@ void test_file_open(void)
...
+	ret = fs_open(&filep, "/", FS_O_READ);
+	zassert_not_equal(ret, 0, "Open a file with wrong path");
+
+	TC_PRINT("\nOpen a file with wrong path\n");
+	ret = fs_open(&filep, "test_file.txt", FS_O_READ);
 	zassert_not_equal(ret, 0, "Open a file with wrong path");
 
+	TC_PRINT("\nOpen a file with wrong abs path\n");
+	ret = fs_open(&filep, "/test_file.txt", FS_O_READ);
+	zassert_not_equal(ret, 0, "Open a file with wrong abs path");
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:413]: Reorder these so that there would be no need for two "Open a file with wrong abs path" "headers".

**Review by de-nordic — APPROVED** [review]
Some comments left. Generally looks OK.

Thanks!
